src/main/java/de/retest/web/selenium/RecheckDriver.java

```diff
@@ -0,0 +1,273 @@
+			throw new IllegalStateException(
+					"You must use the UnbreakableRecheckImpl and first check the state before being able to use the retestId locator." );
+		}
+		final Element searchedFor = by.findElement( lastExpectedState, lastActualState );
+		return wrapped.findElement( By.xpath( searchedFor.getIdentifyingAttributes().getPath() ) );
+	}
+
+	public WebElement findElementByRecheckId( final String recheckId ) {
+		return findElement( new ByRetestId( recheckId ) );
+	}
+
+	@Override
+	public WebElement findElement( final By by ) {
+		if ( by instanceof ByRetestId ) {
+			return findElement( (ByRetestId) by );
+		}
+		// TODO We can implement to find the element with the given value in the lastCheckedState
+		// Should it differ, we can use the new value in the current state and issue a warning!
+		return wrapped.findElement( by );
+	}
+
+	@Override
+	public List<WebElement> findElements( final By by ) {
+		return wrapped.findElements( by );
+	}
+
+	@Override
+	public WebElement findElementById( final String using ) {
+		return ((FindsById) wrapped).findElementById( using );
+	}
+
+	@Override
+	public List<WebElement> findElementsById( final String using ) {
+		return ((FindsById) wrapped).findElementsById( using );
+	}
+
+	@Override
+	public WebElement findElementByClassName( final String using ) {
+		return ((FindsByClassName) wrapped).findElementByClassName( using );
+	}
+
+	@Override
+	public List<WebElement> findElementsByClassName( final String using ) {
+		return ((FindsByClassName) wrapped).findElementsByClassName( using );
+	}
+
+	@Override
+	public WebElement findElementByLinkText( final String using ) {
+		return ((FindsByLinkText) wrapped).findElementByLinkText( using );
+	}
+
+	@Override
+	public List<WebElement> findElementsByLinkText( final String using ) {
+		return ((FindsByLinkText) wrapped).findElementsByLinkText( using );
+	}
+
+	@Override
+	public WebElement findElementByPartialLinkText( final String using ) {
+		return ((FindsByLinkText) wrapped).findElementByPartialLinkText( using );
+	}
+
+	@Override
+	public List<WebElement> findElementsByPartialLinkText( final String using ) {
+		return ((FindsByLinkText) wrapped).findElementsByPartialLinkText( using );
+	}
+
+	@Override
+	public WebElement findElementByName( final String using ) {
+		return ((FindsByName) wrapped).findElementByName( using );
+	}
+
+	@Override
+	public List<WebElement> findElementsByName( final String using ) {
+		return ((FindsByName) wrapped).findElementsByName( using );
+	}
+
+	@Override
+	public WebElement findElementByCssSelector( final String using ) {
+		return ((FindsByCssSelector) wrapped).findElementByCssSelector( using );
+	}
+
+	@Override
+	public List<WebElement> findElementsByCssSelector( final String using ) {
+		return ((FindsByCssSelector) wrapped).findElementsByCssSelector( using );
+	}
+
+	@Override
+	public WebElement findElementByTagName( final String using ) {
+		return ((FindsByTagName) wrapped).findElementByTagName( using );
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Remove this use of "FindsByTagName"; it is deprecated. ([squid:CallToDeprecatedMethod](https://sonarcloud.io/organizations/retest/rules#rule_key=squid%3ACallToDeprecatedMethod))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=de.retest%3Arecheck-web&pullRequest=77&issues=AWaHMZUnlB6-zLfzofGW&open=AWaHMZUnlB6-zLfzofGW)

[ISSUE_KEY]: AWaHMZUnlB6-zLfzofGW